xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -505,15 +505,32 @@ bool CProcessorHD::Render(CRect src, CRect dst, ID3D11Resource* target, CRenderB
   if (SUCCEEDED(m_pVideoContext.As(&videoCtx1)))
   {
     const DXGI_COLOR_SPACE_TYPE source_color = GetDXGIColorSpace(views[2], m_bSupportHDR10);
-    const DXGI_COLOR_SPACE_TYPE target_color = DX::Windowing()->UseLimitedColor() 
-                                               ? DXGI_COLOR_SPACE_RGB_STUDIO_G22_NONE_P709 
-                                               : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+    DXGI_COLOR_SPACE_TYPE target_color;
+
+    if (DX::Windowing()->Is_10bitSwapchain() && (views[2]->hasLightMetadata || views[2]->primaries == AVCOL_PRI_BT2020))
+    {
+      target_color = DX::Windowing()->UseLimitedColor()
+                             ? DXGI_COLOR_SPACE_RGB_STUDIO_G2084_NONE_P2020
+                             : DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020;
+    }
+    else
+    {
+      target_color = DX::Windowing()->UseLimitedColor()
+                             ? DXGI_COLOR_SPACE_RGB_STUDIO_G22_NONE_P709
+                             : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+    }
 
     videoCtx1->VideoProcessorSetStreamColorSpace1(m_pVideoProcessor.Get(), DEFAULT_STREAM_INDEX, source_color);
     videoCtx1->VideoProcessorSetOutputColorSpace1(m_pVideoProcessor.Get(), target_color);
     // makes target available for processing in shaders
     videoCtx1->VideoProcessorSetOutputShaderUsage(m_pVideoProcessor.Get(), 1);
 
+//  Disabled because causes HDR image to dark.
+//  It is also not needed because it is only for dynamic HDR metadata (HDR10+).
+//  This code runs continuously during playback and this is only necessary for HDR10+.
+//  Static HDR metadata is configured only at the beginning.
+//  Probably there is some error with the change of scale (factor 50000) and data types conversions.
+/*
```

Comment:
If it doesn't work, just commenting it out won't get merged I'm sure. Either it needs to be fixed to work, or dropped. However, I highly doubt dropping it would be an option.

Reply:
This code is not necessary for HDR10 static metadata and for HDR10+ is buggy (code or Windows API). For current status (support only HDR10) can be dropped.

Reply:
As I said, either need to fix the fault, or drop it. @afedchin may have some insight on where he got to with that code block.

Reply:
just disable this block is hw HDR10 is used. this block is needed to convert HDR movie to SDR by a driver.

Reply:
I found a better solution for this also. Comment it later.